ThunderCloud/StormLanguageController.swift

```diff
@@ -0,0 +1,410 @@
+                    }
+                }
+            }
+        }
+        
+        return (regionalLanguagePack: regionalLanguagePack, majorLanguagePack: majorLanguagePack)
+    }
+    
+    /// Reloads the language pack based on user preferences and assigns it to the language dictionary
+    public func reloadLanguagePack() {
+
+        //Load languages
+        var finalLanguage = [AnyHashable: Any]()
+        
+        let packs = languagePacks()
+        
+        //Major
+        let majorPack = packs?.majorLanguagePack
+        
+        if let _majorFileName = majorPack?.fileName, let majorPackPath = ContentController.shared.fileUrl(forResource: _majorFileName, withExtension: "json", inDirectory: "languages") {
+            
+            currentLanguage = _majorFileName
+            
+            let majorLanguageDictionary = languageDictionary(for: majorPackPath.path)
+            
+            if let _majorLanguageDictionary = majorLanguageDictionary {
+                
+                for (key, value) in _majorLanguageDictionary {
+                    finalLanguage[key] = value as AnyObject
+                }
+            }
+        }
+        
+        //Minor
+        let minorPack = packs?.regionalLanguagePack
+        if let _minorFileName = minorPack?.fileName, let minorPackPath = ContentController.shared.fileUrl(forResource: _minorFileName, withExtension: "json", inDirectory: "languages") {
+            
+            currentLanguage = _minorFileName
+            
+            let minorLanguageDictionary = languageDictionary(for: minorPackPath.path)
+            
+            if let _minorLanguageDictionary = minorLanguageDictionary as? [String: String] {
+                
+                for (key, value) in _minorLanguageDictionary {
+                    finalLanguage[key] = value as AnyObject
+                }
+            }
+        }
+        
+        //Fall back to default if we need it
+        if finalLanguage.count == 0 {
+            
+            if let appFileURL = ContentController.shared.fileUrl(forResource: "app", withExtension: "json", inDirectory: nil) {
+            
+                let appJSON = try? JSONSerialization.jsonObject(withFile:appFileURL.path, options: [])
+            
+                if let _appJSON = appJSON as? [String: AnyObject], let packString = _appJSON["pack"] as? String, let packURL = URL(string: packString) {
+                    
+                    guard let fileName = packURL.lastPathComponent.components(separatedBy: ".").first, let fullFilePath = ContentController.shared.fileUrl(forResource: fileName, withExtension: "json", inDirectory: "languages") else {
+                        return
+                    }
+                    
+                    self.languageDictionary = languageDictionary(for: fullFilePath.path)
+                    return
+                }
+            }
+        }
+        
```

Comment:
I'm not sure tbh :/ It's just a choice at this stage I guess, if the publish is already borked does it matter?